src/mongocxx/validation_criteria.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mongocxx/validation_criteria.hpp>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/stream/helpers.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+void validation_criteria::rule(bsoncxx::document::value rule) {
+    _owned_rule = std::move(rule);
+    _rule = _owned_rule->view();
+}
+
+void validation_criteria::rule(bsoncxx::document::view rule) {
+    _rule = std::move(rule);
+    if (_owned_rule) {
```

Comment:
It could happen if you set a rule more than once:
```
validation_criteria criteria;
criteria.rule(some_rule); // the first rule is a document::value
...
criteria.rule(some_other_rule); // the new rule is a document::view
```
We don't do anything to prevent users from doing this.  In some cases it might be useful for them to do it?  Like, maybe they have some set of validation criteria around, and change the rule or level in reaction to some event happening or something?